ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala

```diff
@@ -218,13 +231,111 @@ private class JdbcLedgerDao(
       configuration: Configuration,
       rejectionReason: Option[String],
   )(implicit loggingContext: LoggingContext): Future[PersistenceResponse] =
-    throw new UnsupportedOperationException("not supported") // TODO add support
+    withEnrichedLoggingContext(Logging.submissionId(submissionId)) { implicit loggingContext =>
+      logger.info("Storing a configuration entry")
+      dbDispatcher.executeSql(
+        metrics.daml.index.db.storeConfigurationEntryDbMetrics
+      ) { implicit conn =>
+        val optCurrentConfig = ParametersTable.getLedgerEndAndConfiguration(conn)
+        val optExpectedGeneration: Option[Long] =
+          optCurrentConfig.map { case (_, c) => c.generation + 1 }
+        val finalRejectionReason: Option[String] =
+          optExpectedGeneration match {
+            case Some(expGeneration)
+                if rejectionReason.isEmpty && expGeneration != configuration.generation =>
+              // If we're not storing a rejection and the new generation is not succ of current configuration, then
+              // we store a rejection. This code path is only expected to be taken in sandbox. This follows the same
+              // pattern as with transactions.
+              Some(
+                s"Generation mismatch: expected=$expGeneration, actual=${configuration.generation}"
+              )
+
+            case _ =>
+              // Rejection reason was set, or we have no previous configuration generation, in which case we accept any
+              // generation.
+              rejectionReason
+          }
+
+        val update = finalRejectionReason match {
+          case None =>
+            Update.ConfigurationChanged(
+              recordTime = Time.Timestamp.assertFromInstant(recordedAt),
+              submissionId =
+                submissionId.asInstanceOf[SubmissionId], // TODO append-only: FIXME type cast
+              participantId = null.asInstanceOf[
+                v1.ParticipantId
+              ], // TODO append-only: FIXME type cast // not used for DBDTO generation
```

Comment:
(unrelated to this PR) Which begs the question, why does the field exist and why is it not indexed?

@mziolekda Do you know what the purpose of [ConfigurationChanged.participantId](https://github.com/digital-asset/daml/blob/main/ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/Update.scala#L34) is? This information is not stored in the index database, and is not used in the ledger API.